homeassistant/helpers/target.py

```diff
@@ -238,3 +251,102 @@ def async_extract_referenced_entity_ids(
     )
 
     return selected
+
+
+class TargetStateChangeTracker:
+    """Helper class to manage state change tracking for targets."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        selector_data: TargetSelectorData,
+        job_type: HassJobType | None,
```

Comment:
This is an optimization to avoid checking the type of the callback when the same callback is used for a big number of listeners. I suggest to leave it out for now until we know it's needed.